src/test/regress/expected/multi_mx_router_planner_0.out

```diff
@@ -0,0 +1,1514 @@
+ 41 |         1 | aznavour     |      11814
+(5 rows)
+
+-- parametric prepare queries can be router plannable
+PREPARE author_articles(int) as
+	SELECT *
+	FROM articles_hash_mx
+	WHERE author_id = $1;
+EXECUTE author_articles(1);
+DEBUG:  Creating router plan
+DEBUG:  query has a single distribution column value: 1
+ id | author_id |    title     | word_count
+---------------------------------------------------------------------
+  1 |         1 | arsenous     |       9572
+ 11 |         1 | alamo        |       1347
+ 21 |         1 | arcading     |       5890
+ 31 |         1 | athwartships |       7271
+ 41 |         1 | aznavour     |      11814
+(5 rows)
+
+-- queries inside plpgsql functions could be router plannable
+SET citus.enable_metadata_sync TO OFF;
+CREATE OR REPLACE FUNCTION author_articles_max_id() RETURNS int AS $$
+DECLARE
+  max_id integer;
+BEGIN
+	SELECT MAX(id) FROM articles_hash_mx ah
+		WHERE author_id = 1
+		into max_id;
+	return max_id;
+END;
+$$ LANGUAGE plpgsql;
+SELECT author_articles_max_id();
+DEBUG:  Creating router plan
+CONTEXT:  SQL statement "SELECT MAX(id) FROM articles_hash_mx ah
+		WHERE author_id = 1"
+PL/pgSQL function author_articles_max_id() line XX at SQL statement
+DEBUG:  query has a single distribution column value: 1
+CONTEXT:  SQL statement "SELECT MAX(id) FROM articles_hash_mx ah
+		WHERE author_id = 1"
+PL/pgSQL function author_articles_max_id() line XX at SQL statement
+ author_articles_max_id
+---------------------------------------------------------------------
+                     41
+(1 row)
+
+-- plpgsql function that return query results are not router plannable
+CREATE OR REPLACE FUNCTION author_articles_id_word_count() RETURNS TABLE(id bigint, word_count int) AS $$
+DECLARE
+BEGIN
+	RETURN QUERY
+		SELECT ah.id, ah.word_count
+		FROM articles_hash_mx ah
+		WHERE author_id = 1;
+
+END;
+$$ LANGUAGE plpgsql;
+SELECT * FROM author_articles_id_word_count();
+DEBUG:  Creating router plan
+CONTEXT:  SQL statement "SELECT ah.id, ah.word_count
+		FROM articles_hash_mx ah
+		WHERE author_id = 1"
+PL/pgSQL function author_articles_id_word_count() line XX at RETURN QUERY
+DEBUG:  query has a single distribution column value: 1
+CONTEXT:  SQL statement "SELECT ah.id, ah.word_count
+		FROM articles_hash_mx ah
+		WHERE author_id = 1"
+PL/pgSQL function author_articles_id_word_count() line XX at RETURN QUERY
+ id | word_count
+---------------------------------------------------------------------
+  1 |       9572
+ 11 |       1347
+ 21 |       5890
+ 31 |       7271
+ 41 |      11814
+(5 rows)
+
+RESET citus.enable_metadata_sync;
+-- materialized views can be created for router plannable queries
+CREATE MATERIALIZED VIEW mv_articles_hash_mx AS
+	SELECT * FROM articles_hash_mx WHERE author_id = 1;
+DEBUG:  Creating router plan
+DEBUG:  query has a single distribution column value: 1
+DEBUG:  drop auto-cascades to type public.pg_temp_25895
+DEBUG:  drop auto-cascades to type public.pg_temp_25895[]
```

Comment:
I feel like these pg_temp suffixes might become flaky in the future.

So, should we maybe wrap CREATE MAT VIEW as below?

```sql
SET client_min_messages TO WARNING;
CREATE MATERIALIZED VIEW ...
SET client_min_messages TO <whatever value it was set to before>
```

Reply:
Oh, but we also probably want to know that the command invokes router planner. So, maybe we should add a output normalization rule for this into normalize.sed so that we can replace any occurrences of `DEBUG: drop auto-cascades to type public.pg_temp_\d+` with `DEBUG: drop auto-cascades to type public.pg_temp_XXX` or such.

That way, we can also get rid of this extra output file I think.

I suggest reaching out to @naisila if you have any questions on how to do that.